virtcontainers/sandbox.go

```diff
@@ -926,6 +929,102 @@ func (s *Sandbox) removeNetwork() error {
 	return s.network.remove(s, s.networkNS, s.networkNS.NetNsCreated)
 }
 
+// UpdateRoute updates the sandbox route table (e.g. for portmapping support).
+func (s *Sandbox) UpdateRoute(routes []*grpc.Route) error {
+	if err := s.agent.updateRoutes(routes); err != nil {
+		return err
+	}
+	s.networkNS.Routes = routes
+	return s.storage.storeSandboxNetwork(s.id, s.networkNS)
+}
+
+// AddNetwork adds new nic to the sandbox.
+func (s *Sandbox) AddNetwork(inf grpc.Interface) error {
```

Comment:
@caoruidong You will need to call AddNetwork after entering the network namespace with DoNetNs()

Reply:
Some people worry about overhead about veth pair in #113 .  I will update this PR if most people agree on the veth pair way.

Reply:
Yes thanks @caoruidong, let's have the "standard" network hotplug first and move to some optimizations/improvements later (we'll make this configurable I think).